lib/mirage.ml

```diff
@@ -471,10 +472,14 @@ module Entropy = struct
     append_main "module %s = %s" (module_name t) (construction t) ;
     newline_main () ;
     append_main "let %s () =" (name t);
-    append_main "  %s.connect %s" (module_name t) (id t);
+    append_main "  %s.connect %s >>= function" (module_name t) (id t);
+    append_main "  | `Error e -> %s" (driver_initialisation_error "entropy");
+    append_main "  | `Ok entropy ->";
+    append_main "  %s.handler entropy Nocrypto.Rng.Accumulator.add_rr >>= fun () ->" (module_name t);
```

Comment:
Whatever the solution, it would be nice if we could get rid of the global mutable variable `Rng.Accumulator.acc`. It's very useful to see all sources of non-determinism in a program. As it is, there is no visible link between `entropy` and `tls` in the generated `main.ml`.